### ISSUE TITLE
New Salzinnes mei files

### ISSUE THREAD

**Issue by napulen** [DDMAL/cantus#304]
@fujinaga wants to demo the new Salzinnes MEI files.

The steps I've got so far:

- I got the mei files from the `split_text_aligned_mei, new encoding method (7-24)` folder in the share
- Uploaded those files to the *staging* deployment of CU in CC
- Without spending too much time looking into the files, I put them in the right location and tried to import them via ```python python manage.py import_mei_data mei_to_solr salzinnes```

That didn't work. The mei files we have in CU are an old version of MEI, labeled as `meiversion="2012"`. The new files are `meiversion=4.0.0`, however, that doesn't seem to be a problem yet. It seems that the script is importing the MEI because it isn't finding the Salzinnes manuscript.

I took a look and see what we know about the procedure for adding OMR data, but it is not really mentioned in the `cantus-documentation` repo or anywhere else. All we have for importing OMR data, as far as I have seen, is this line in the `README`:

```python
# Import OMR Data

./manage.py import_mei_data mei_to_solr salzinnes
# Repeat the above with 'st_gallen_390' and then 'st_gallen_391' instead of 'salzinnes'
```
... and what is mentioned in issue #42 and a few others.

I took a look at the implementation of `import_mei_data`, the ids of the manuscripts are hardcoded to the ones the manuscripts have in the production website:

```python
if manuscript == "salzinnes":
            self.stdout.write("Salzinnes manuscript selected.")
            siglum = "cdn-hsmu-m2149l4"
            id = 133
            mei_location = "data_dumps/mei/salz"
            csv_location = "data_dumps/mei_csv/salzinnes.csv"
        elif manuscript == "st_gallen_390":
            self.stdout.write("St. Gallen 390 manuscript selected.")
            siglum = "ch-sgs-390"
            id = 127
            mei_location = "data_dumps/mei/csg-390"
            csv_location = "data_dumps/mei_csv/csg-390.csv"
        elif manuscript == "st_gallen_391":
            self.stdout.write("St. Gallen 391 manuscript selected.")
            siglum = "ch-sgs-391"
            id = 128
            mei_location = "data_dumps/mei/csg-391"
            csv_location = "data_dumps/mei_csv/csg-391.csv"
```

This means our scripts don't work outside the production environment (e.g., the staging version). In any case, we can't keep hardcoding ids in the management scripts. These need to be re-written.

In short, importing OMR data into Cantus Ultimus is not straightforward as of now and it requires us to:

- Re-write our MEI importing tools
- Check what needs to be changed in order to incorporate `meiversion=4.0.0` files
- Reproduce that workflow outside the production environment

What steps do you recommend?  @vigliensoni @fujinaga

**Comment by vigliensoni** [comment]
We could use part of the mechanism we have for the mapping tool. At that stage, the user enters the location of CSV and MEI files. From these, we can get the values needed here (i.e., `siglum`, `id`, `mei_location`, and `csv_location`)

**Comment by napulen** [comment]
As far as I remember, at the mapping tool, the users enter the location of the CSV (Cantus Database) and the IIIF Manifest, not the MEI files.

**Comment by vigliensoni** [comment]
Yes, but what about if we use a similar mechanism to ingest these files and not hardcode the values?

Also, how feasible do you think it would be support MEI 3.0 and 4.0?
